mailbox/store/src/test/java/org/apache/james/mailbox/store/AbstractMessageIdManagerStorageTest.java

```diff
@@ -0,0 +1,280 @@
+
+        messageIdManager.setInMailboxes(messageId, ImmutableList.of(mailbox2.getMailboxId()), session);
+
+        MessageResult messageResult2 = FluentIterable
+            .from(messageIdManager.getMessages(ImmutableList.of(messageId), FetchGroupImpl.MINIMAL, session))
+            .filter(new Predicate<MessageResult>() {
+                @Override
+                public boolean apply(MessageResult input) {
+                    return input.getMailboxId().equals(mailbox1.getMailboxId());
+                }
+            })
+            .toList()
+            .get(0);
+        MessageUid messageUid2 = messageResult2.getUid();
+        long modSeq2 = messageResult2.getModSeq();
+
+        assertThat(messageUid1).isEqualTo(messageUid2);
+        assertThat(modSeq1).isEqualTo(modSeq2);
+    }
+
+    @Test
+    public void deleteMessageShouldRemoveMessageFromMailbox() throws Exception {
+        MessageId messageId = testingData.persist(mailbox1.getMailboxId(), FLAGS);
+
+        messageIdManager.delete(messageId, ImmutableList.of(mailbox1.getMailboxId()), session);
+
+        assertThat(messageIdManager.getMessages(ImmutableList.of(messageId), FetchGroupImpl.MINIMAL, session)).isEmpty();
+    }
+
+    @Test
+    public void deleteMessageShouldRemoveMessageOnlyFromMailbox() throws Exception {
+        MessageId messageId = testingData.persist(mailbox1.getMailboxId(), FLAGS);
+        messageIdManager.setInMailboxes(messageId, ImmutableList.of(mailbox2.getMailboxId()), session);
+
+        messageIdManager.delete(messageId, ImmutableList.of(mailbox1.getMailboxId()), session);
+
+        List<MessageResult> messageResults = messageIdManager.getMessages(ImmutableList.of(messageId), FetchGroupImpl.MINIMAL, session);
+        assertThat(messageResults).hasSize(1);
+        assertThat(messageResults.get(0).getMailboxId()).isEqualTo(mailbox2.getMailboxId());
+    }
+
+    @Test
+    public void deleteMessageShouldNotRemoveMessageOnAnotherMailbox() throws Exception {
+        MessageId messageId = testingData.persist(mailbox1.getMailboxId(), FLAGS);
+
+        messageIdManager.delete(messageId, ImmutableList.of(mailbox2.getMailboxId()), session);
+
+        List<MessageResult> messageResults = messageIdManager.getMessages(ImmutableList.of(messageId), FetchGroupImpl.MINIMAL, session);
+        assertThat(messageResults).hasSize(1);
+        assertThat(messageResults.get(0).getMailboxId()).isEqualTo(mailbox1.getMailboxId());
+    }
+
+    @Test
+    public void setFlagsShouldUpdateFlags() throws Exception {
+        Flags newFlags = new Flags(Flags.Flag.SEEN);
+        MessageId messageId = testingData.persist(mailbox1.getMailboxId(), FLAGS);
+
+        messageIdManager.setFlags(newFlags, MessageManager.FlagsUpdateMode.ADD, messageId, session);
+
+        MessageResult messageResult = messageIdManager.getMessages(ImmutableList.of(messageId), FetchGroupImpl.MINIMAL, session).get(0);
+        assertThat(messageResult.getFlags()).isEqualTo(newFlags);
+    }
```

Comment:
Maybe you can add a test to match the contract that changing a flag change it on all its mailboxes (or not ^^).